pkg/manifests/manifests.go

```diff
@@ -3896,21 +3903,31 @@ func (f *Factory) TelemeterClientServiceAccount() (*v1.ServiceAccount, error) {
 	return s, nil
 }
 
+func (f *Factory) TelemeterClientSecretTemplate() (*v1.Secret, error) {
+	return f.NewSecret(f.assets.MustNewAssetReader(TelemeterClientSecret))
+}
+
 // TelemeterClientSecret generates a new Secret for Telemeter client.
-func (f *Factory) TelemeterClientSecret() (*v1.Secret, error) {
-	s, err := f.NewSecret(f.assets.MustNewAssetReader(TelemeterClientSecret))
+func (f *Factory) TelemeterClientSecret(secretData map[string][]byte) (*v1.Secret, error) {
+	s, err := f.TelemeterClientSecretTemplate()
 	if err != nil {
 		return nil, err
 	}
 
+	if f.config.ClusterMonitoringConfiguration.TelemeterClientConfig.Token != "" {
```

Comment:
I'd rather move the check to the task and decide there whether we need to update the secret or not (e.g. if the existing token value is equal to the configured token then do nothing otherwise update the secret). This would leave this method untouched.